show-grid/src/main/java/com/esri/arcgismaps/sample/showgrid/MainActivity.kt

```diff
@@ -0,0 +1,327 @@
+     * and handles behavior for when a label position is selected.
+     */
+    private fun setupLabelPositionDropdown(popupDialogBinding: PopupDialogBinding) {
+        popupDialogBinding.labelPositionDropdown.apply {
+            setAdapter(ArrayAdapter(
+                applicationContext,
+                android.R.layout.simple_list_item_1,
+                resources.getStringArray(R.array.positions_array))
+            )
+
+            onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
+                // set the label position
+                labelPosition = when (position) {
+                    0 -> GridLabelPosition.AllSides
+                    1 -> GridLabelPosition.BottomLeft
+                    2 -> GridLabelPosition.BottomRight
+                    3 -> GridLabelPosition.Center
+                    4 -> GridLabelPosition.Geographic
+                    5 -> GridLabelPosition.TopLeft
+                    6 -> GridLabelPosition.TopRight
+                    else -> return@OnItemClickListener showError("Unsupported option")
+                }
+                changeLabelPosition(labelPosition)
+            }
+        }
+    }
+
+    /**
+     * Sets up the [popupDialogBinding] for the checkbox making labels visible or invisible.
+     */
+    private fun setupLabelsCheckbox(popupDialogBinding: PopupDialogBinding) {
+        popupDialogBinding.labelsCheckBox.apply {
+            isChecked = true
+            // hide and show label visibility when the checkbox is clicked
+            setOnClickListener {
+                isLabelVisible = isChecked
+                setLabelVisibility(isLabelVisible)
+            }
+        }
+    }
+
+    /**
+     * Sets the labels visibility based on [isVisible].
+     */
+    private fun setLabelVisibility(isVisible: Boolean) {
+        val grid = mapView.grid ?: return
+        grid.labelVisibility = isVisible
+    }
+
+    /**
+     * Sets the [color] of the grid lines.
+     */
+    private fun changeGridColor(color: Color) {
+        val grid = mapView.grid ?: return
+        val gridLevels = grid.levelCount
+        for (gridLevel in 0 until gridLevels) {
+            val lineSymbol: LineSymbol =
+                SimpleLineSymbol(SimpleLineSymbolStyle.Solid, color, (gridLevel + 1).toFloat())
+            grid.setLineSymbol(gridLevel, lineSymbol)
+        }
+    }
+
+    /**
+     * Sets the [labelColor] of the labels on the grid.
+     */
+    private fun changeLabelColor(labelColor: Color) {
+        val grid = mapView.grid ?: return
+        val gridLevels = grid.levelCount
+        for (gridLevel in 0 until gridLevels) {
+            val textSymbol = TextSymbol().apply {
+                color = labelColor
+                size = 14f
+                horizontalAlignment = HorizontalAlignment.Left
+                verticalAlignment = VerticalAlignment.Bottom
+                haloColor = Color.white
```

Comment:
In noticed that for some grid types e.g. USNG the labels don't look great when the label colour is white. I think it's because this outline colour is also white. Maybe when the label colour is white the outline could be set to black?